### ISSUE TITLE
OS X build error with nodes v6.9.1

### ISSUE THREAD

**Issue by littleskunk** [LinusU/fs-xattr#17]
OSX 14.5.0
Nodejs v6.9.1 (LTS)

First try errored out
```
skunks-Mac-mini:~ skunk$ git clone https://github.com/LinusU/fs-xattr
Cloning into 'fs-xattr'...
remote: Counting objects: 184, done.
remote: Total 184 (delta 0), reused 0 (delta 0), pack-reused 184
Receiving objects: 100% (184/184), 31.68 KiB | 0 bytes/s, done.
Resolving deltas: 100% (96/96), done.
skunks-Mac-mini:~ skunk$ cd fs-xattr/
skunks-Mac-mini:fs-xattr skunk$ npm install
npm WARN deprecated to-iso-string@0.0.2: to-iso-string has been deprecated, use @segment/to-iso-string instead.
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue

> fs-xattr@0.1.14 install /Users/skunk/fs-xattr
> node-gyp rebuild

gyp WARN install got an error, rolling back install
gyp ERR! configure error 
gyp ERR! stack Error: read ECONNRESET
gyp ERR! stack     at exports._errnoException (util.js:1026:11)
gyp ERR! stack     at TLSWrap.onread (net.js:569:26)
gyp ERR! System Darwin 14.5.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/skunk/fs-xattr
gyp ERR! node -v v6.9.1
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok 

npm ERR! Darwin 14.5.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.9
npm ERR! code ELIFECYCLE
npm ERR! fs-xattr@0.1.14 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the fs-xattr@0.1.14 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the fs-xattr package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs fs-xattr
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls fs-xattr
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/skunk/fs-xattr/npm-debug.log
```

Second try
```
skunks-Mac-mini:fs-xattr skunk$ npm install

> fs-xattr@0.1.14 install /Users/skunk/fs-xattr
> node-gyp rebuild

  CXX(target) Release/obj.target/xattr/src/async.o
  CXX(target) Release/obj.target/xattr/src/error.o
  CXX(target) Release/obj.target/xattr/src/sync.o
../src/sync.cc:46:59: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
  v8::Local<v8::Object> actualBuffer = bufferConstructor->NewInstance(3, constructorArgs);
                                                          ^
/Users/skunk/.node-gyp/6.9.1/include/node/v8.h:3270:31: note: 'NewInstance' has been explicitly marked deprecated here
                Local<Object> NewInstance(int argc, Local<Value> argv[]) const);
                              ^
/Users/skunk/.node-gyp/6.9.1/include/node/v8config.h:336:3: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
  ^
1 warning generated.
  CXX(target) Release/obj.target/xattr/src/util.o
  CXX(target) Release/obj.target/xattr/src/xattr.o
  SOLINK_MODULE(target) Release/xattr.node
```

**Comment by LinusU** [comment]
Seems like there was a network problem with your first try. The second try is only showing a warning, but it should have worked?

We should fix the warning though, I think we just need to replace `blah->NewInstance(` with `Nan::NewInstance(blah, `

**Comment by littleskunk** [comment]
I don't think it is a network Problem. I try to build electron binaries and only this package has "network problems". It can reproduce it. I start my build server and the first try always fails. I think it is a side effect of the second problem. Lets focus on the second problem.

The solution looks good. I can test it later.

How about travis update? nodejs v6.0.0 is not up to date.

**Comment by LinusU** [comment]
The second problem is now fixed and released as 0.1.15 :) please reopen if there is still problems
